Add continous JSON Logging monitor

eknoes opened the pull request:
Add a monitor that appends a single-line JSON object (https://jsonlines.org/) containing the current state every 60 seconds. I want to use it so that I can do some time-related analysis, e.g. how much coverage is found after a certain amount of time etc.

domenukk commented:
That looks useful!
A few thoughts:
- Is there a reason you're re-opening every time? It's only a single syscall per minute, so doesn't matter much, but still seems redundant, you could just keep the file open
- Especially in the beginning of fuzzing, you usually would be interested in sub-minute stats, so it could make sense to be able to set the seconds, or even have a closure to decide if it should be stored or not. So like, exponentially wait longer, only store if there's new coverage, etc.

eknoes commented:
I like the idea with the closure and just implemented that! Honestly, I just re-opened it all the time because the TOMLMonitor did the same - just figured out thats because `Clone` is needed for the Monitor but not implemented on File, so I reverted that again.

domenukk commented:
Sweet, thanks! :)
